apps/arena/lib/arena/game_updater.ex

```diff
@@ -924,34 +985,71 @@ defmodule Arena.GameUpdater do
 
     {
       Map.put(projectiles_acc, projectile.id, projectile),
-      Map.put(players_acc, player.id, player)
+      Map.put(players_acc, player.id, player),
+      crate_acc
+    }
+  end
+
+  # Projectile collided a crate
+  defp process_projectile_collision(
+         projectile,
+         %{category: :crate} = crate,
+         _,
+         _,
+         {projectiles_acc, players_acc, crates_acc}
+       )
+       when not is_nil(crate) do
```

Comment:
This guard is useless, if the pattern `%{category: :crate} = crate` matches then it can't ever be `nil`

Reply:
you're right! the player pattern match had something similar so i deleted that too